src/pybind/mgr/cephadm/agent.py

```diff
@@ -44,212 +42,89 @@ def cherrypy_filter(record: logging.LogRecord) -> int:
-
-    def configure_cherrypy(self) -> None:
-        cherrypy.config.update({
-            'environment': 'production',
-            'server.socket_host': self.server_addr,
-            'server.socket_port': self.server_port,
-            'engine.autoreload.on': False,
-            'server.ssl_module': 'builtin',
-            'server.ssl_certificate': self.cert_tmp.name,
-            'server.ssl_private_key': self.key_tmp.name,
-        })
+        self.host_data: Server = None
+
+    def configure_routes(self) -> None:
+
+        self.host_data = HostData(self.mgr,
+                                  self.server_port,
+                                  self.server_addr,
+                                  self.cert_file.name,
+                                  self.key_file.name)
 
         # configure routes
-        root = Root(self.mgr)
-        host_data = HostData(self.mgr)
         d = cherrypy.dispatch.RoutesDispatcher()
-        d.connect(name='index', route='/', controller=root.index)
-        d.connect(name='sd-config', route='/prometheus/sd-config', controller=root.get_sd_config)
-        d.connect(name='rules', route='/prometheus/rules', controller=root.get_prometheus_rules)
-        d.connect(name='host-data', route='/data', controller=host_data.POST,
+        d.connect(name='host-data', route='/',
+                  controller=self.host_data.POST,
                   conditions=dict(method=['POST']))
 
-        conf = {'/': {'request.dispatch': d}}
-        cherrypy.tree.mount(None, "/", config=conf)
+        cherrypy.tree.mount(None, '/data', config={'/': {'request.dispatch': d}})
 
-    def run(self) -> None:
+    def configure_tls(self) -> None:
         try:
+            old_creds = self.mgr.get_store('cephadm_endpoint_credentials')
+            if not old_creds:
+                raise OrchestratorError('No old credentials for cephadm endpoint found')
+            old_creds_dict = json.loads(old_creds)
+            old_key = old_creds_dict['key']
+            old_cert = old_creds_dict['cert']
+            self.ssl_certs.load_root_credentials(old_cert, old_key)
+        except (OrchestratorError, json.decoder.JSONDecodeError, KeyError, ValueError):
+            self.ssl_certs.generate_root_cert(self.mgr.get_mgr_ip())
+
+        cert, key = self.ssl_certs.generate_cert(self.mgr.get_mgr_ip())
+        self.key_file = tempfile.NamedTemporaryFile()
+        self.key_file.write(key.encode('utf-8'))
+        self.key_file.flush()  # pkey_tmp must not be gc'ed
+        self.cert_file = tempfile.NamedTemporaryFile()
+        self.cert_file.write(cert.encode('utf-8'))
+        self.cert_file.flush()  # cert_tmp must not be gc'ed
+        verify_tls_files(self.cert_file.name, self.key_file.name)
+
+        cephadm_endpoint_creds = {
+            'cert': self.ssl_certs.get_root_cert(),
+            'key': self.ssl_certs.get_root_key()
+        }
+        self.mgr.set_store('cephadm_endpoint_credentials', json.dumps(cephadm_endpoint_creds))
```

Comment:
As I mentioned previously, I'm not a big fan of storing serialized data to a KV store. It makes it harder to deal with that from the CLI, and a KV DB is great at dealing with keys and key hierarchies, so why serializing data from dictionary when KVs are essentially dictionaries? This is not a relational DB.

Reply:
Fixed